### PR TITLE
Modify GameEventPlayer fields name

### LINKED ISSUE #56: Modify GameEventPlayer entity fields
modify the player_id field name to be player and game_event_id name to be game_event

### PR DESCRIPTION
**Pull request by MaayanMashhadi**
modify the fields: game_event_id to game_event
and player_id to player
Also, creating new migration file and change tests.py 

closes #56

### CODE REVIEW

**Review by Danielsio — APPROVED** [review]
Nicely Done Maayan. keep up the good work !

File: game_event/migrations/0001_initial.py
Comment (Danielsio) [inline game_event/migrations/0001_initial.py:1]: I belive this changes (update court_id to court) are already been changed in #55.

is there a specific reason you decided you change it again here ?

File: game_event/migrations/0001_initial.py [hunk 1]
```diff
@@ -31,7 +31,7 @@ class Migration(migrations.Migration):
                                                         ('Cricket', 'Cricket'), ('Handball', 'Handball')],
                                                max_length=100,
                                                validators=[django.core.validators.MinLengthValidator(4)])),
-                ('court_id', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='court.court')),
+                ('court', models.ForeignKey(on_delete=django.db.models.deletion.CASCADE, to='court.court')),
             ],
```
Comment (Danielsio) [inline game_event/migrations/0001_initial.py:35]: I belive this changes (update court_id to court) are already been changed in #55.

is there a specific reason you decided you change it again here ?

File: game_event/models.py [hunk 2]
```diff
@@ -2,18 +2,7 @@
...
-class BallGame(models.TextChoices):
-    Soccer = 'Soccer', 'Soccer'
-    Basketball = 'Basketball', 'Basketball'
-    Volleyball = 'Volleyball', 'Volleyball'
-    Baseball = 'Baseball', 'Baseball'
-    Tennis = 'Tennis', 'Tennis'
-    Rugby = 'Rugby', 'Rugby'
-    Golf = 'Golf', 'Golf'
-    Cricket = 'Cricket', 'Cricket'
-    Handball = 'Handball', 'Handball'
```
Comment (Danielsio) [inline game_event/models.py:16]: you already removed it in #54

File: game_event/tests.py [hunk 3]
```diff
@@ -48,29 +49,29 @@ def test_delete_game_event(self, saved_game_event):
...
                                      min_number_of_players=TEST_MIN, max_number_of_players=TEST_MAX,
-                                     court_id=saved_court, ball_game='').full_clean()
+                                     court=saved_court, ball_game='').full_clean()
 
     def test_create_game_event_with_null_ball_game(self, saved_court):
         with pytest.raises(ValidationError):
             GameEvent.objects.create(id=TEST_ID, time=TEST_TIME, level_of_game=TEST_LEVEL,
                                      min_number_of_players=TEST_MIN, max_number_of_players=TEST_MAX,
-                                     court_id=saved_court,).full_clean()
+                                     court=saved_court,).full_clean()
```
Comment (Danielsio) [inline game_event/tests.py:77]: a lot of those changes are already been applied in #55.

**Review by Danielsio — COMMENTED** [review]
There are quite a few changes here that repeat themselves, I currently recommend leaving only the changes related to game_event_player.

Other then that your code looks great.

**Comment by MaayanMashhadi** [comment]
> There are quite a few changes here that repeat themselves, I currently recommend leaving only the changes related to game_event_player.
> 
> Other then that your code looks great.

You're right, I returned to the original state

**Review by Danielsio — APPROVED** [review]

File: game_event_player/models.py [hunk 4]
```diff
@@ -4,8 +4,8 @@
 
 
 class GameEventPlayer(models.Model):
-    game_event_id = models.ForeignKey(GameEvent, on_delete=models.CASCADE)
-    player_id = models.ForeignKey(Player, on_delete=models.CASCADE)
+    game_event = models.ForeignKey(GameEvent, on_delete=models.CASCADE)
+    player = models.ForeignKey(Player, on_delete=models.CASCADE)
     ball_responsible = models.BooleanField(default=False)
 
     class Meta:
```
Comment (Yuval-Vino) [inline game_event_player/models.py:11]: you should also make the changes in the Unique Together meta

File: game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Generated by Django 4.2 on 2023-04-30 13:12
+
+from django.db import migrations
+
+
+class Migration(migrations.Migration):
```
Comment (Yuval-Vino) [inline game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py:6]: new migration should also deal with unique together constraints update, you can have some reference from here #58

**Review by Yuval-Vino — CHANGES_REQUESTED** [review]

File: game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py (hunk 5 above)
Comment (MaayanMashhadi) [reply game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py:6]: > new migration should also deal with unique together constraints update, you can have some reference from here #58

Hi, Thank you I did it

Comment (Yuval-Vino) [reply game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py:6]: sorry I didn't mention it before, it should be auto-generated by Django, you should not make the changes manually.
was it auto-generated here?

Comment (MaayanMashhadi) [reply game_event_player/migrations/0002_rename_game_event_id_gameeventplayer_game_event_and_more.py:6]: > sorry I didn't mention it before, it should be auto-generated by Django, you should not make the changes manually. was it auto-generated here?

Yes, I did the make migration command for it

**Comment by Danielsio** [comment]
are we still depending on #55 for this PR ?

**Review by SchSeba — APPROVED** [review]
/lgtm

**Comment by yftacherzog** [comment]
> > are we still depending on #55 for this PR ?
> 
> I'm not sure if depending is the right word here... On the tests.py I used game_event fixture, and his court field's name here is still court_id (and #55 deal with it). Cause of that, I think that #55 needs to be merged before this PR

@MaayanMashhadi #55 was merged. You can rebase this one, address the conflicts and make sure everything is done on your end.

**Comment by MaayanMashhadi** [comment]
> > > are we still depending on #55 for this PR ?
> > 
> > 
> > I'm not sure if depending is the right word here... On the tests.py I used game_event fixture, and his court field's name here is still court_id (and #55 deal with it). Cause of that, I think that #55 needs to be merged before this PR
> 
> @MaayanMashhadi #55 was merged. You can rebase this one, address the conflicts and make sure everything is done on your end.

Did it. Just to say: after rebasing, I did make migration to game_event_player again
and the dependencies changed to be: from ('game_event', '0001_initial') to
('game_event', '0002_rename_court_id_gameevent_court').
I think it makes sense, but I want to be sure about it.

**Review by yftacherzog — APPROVED** [review]

**Pull request merged**